src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,243 @@
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member"]
+      },
+      {
+        option: "-p, --properties [properties]"
+      },
+      {
+        option: "-f, --filter [filter]"
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+
+      let users: User[] = [];
+
+      switch (args.options.role) {
+        case 'Owner':
+          users = await this.getOwners(args.options, groupId, logger);
+          break;
+        case 'Member':
+          users = await this.getMembers(args.options, groupId, logger);
+          break;
+        default:
+          const owners = await this.getOwners(args.options, groupId, logger);
+          const members = await this.getMembers(args.options, groupId, logger);
+
+          if (!args.options.properties) {
+            owners.forEach((owner: ExtendedUser) => {
+              for (let i = 0; i < members.length; i++) {
+                if (members[i].userPrincipalName === owner.userPrincipalName) {
+                  if (!owner.roles.includes('Member')) {
+                    owner.roles.push('Member');
+                  }
+                }
+              }
+            });
+          }
+
+          users = owners.concat(members);
+
+          if (!args.options.properties) {
+            users = users.filter((value, index, array) => index === array.findIndex(item => item.userPrincipalName === value.userPrincipalName));
+          }
+      }
+
+      await logger.log(users);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getGroupId(options: Options): Promise<string> {
+    const { groupId, groupDisplayName } = options;
+
+    if (groupId) {
+      return groupId;
+    }
+
+    return await aadGroup.getGroupIdByDisplayName(groupDisplayName!);
+  }
+
+  private async getOwners(options: Options, groupId: string, logger: Logger): Promise<ExtendedUser[]> {
```

Comment:
Aren't the getOwners and getMembers functions like 99% the same?
I suggest we use one function `getUsers(role: 'owner' | 'member', etc)` or something like that.